internal/provider/resource_consul_cluster.go

```diff
@@ -261,6 +272,29 @@ func resourceConsulClusterCreate(ctx context.Context, d *schema.ResourceData, me
 		return diag.Errorf("specified Consul version (%s) is unavailable; must be one of: %v", consulVersion, availableConsulVersions)
 	}
 
+	// If specified, validate and parse the primary link provided for federation.
+	primary_link, ok := d.GetOk("primary_link")
+	var primary *sharedmodels.HashicorpCloudLocationLink
+	if ok {
+		primary, err = parseLinkURL(primary_link.(string), ConsulClusterResourceType)
```

Comment:
You might be able to replace some of this with `buildLinkFromURL()`, eg:
```
buildLinkFromURL(primary_link.(string), ConsulClusterResourceType, client.Config.OrganizationID)
```

Reply:
@roaks3 : So the reason I am not using the `buildLinkFromURL` helper here is because there could be a scenario (planned for future support), where the primary could belong to a different organization than the default org ID (which is client.Config.OrganizationID). Implementing it this way, handles both cases.